core/shared/src/main/scala/matryoshka/patterns/CoEnv.scala

```diff
@@ -22,9 +22,17 @@ import scalaz._, Scalaz._
 
 /** The pattern functor for Free. */
 final case class CoEnv[E, F[_], A](run: E \/ F[A])
+
 object CoEnv extends CoEnvInstances {
   def coEnv[E, F[_], A](v: E \/ F[A]): CoEnv[E, F, A] = CoEnv(v)
 
+  def hmap[F[_], G[_], A](f: F ~> G) =
```

Comment:
“higher-kinded”, as in https://github.com/pa-ba/compdata/blob/master/src/Data/Comp/Multi/HFunctor.hs#L89 (these are just moved from ones we wrote in Quasar).

More of these type classes exist in #28, but the Cats naming scheme might lead us to `FunctorK`, etc. instead. Finally, there is work on adding polykinds to scalac, which would mean that `FunctorK` and `Functor` can be unified into a single type class, and then this operation becomes just `map` … which is actually a bit of a problem, because we have both `HFunctor` _and_ `Functor` on this type currently …